openai/openai.go

```diff
@@ -73,6 +73,7 @@ type ChatCompletionRequest struct {
 	PresencePenalty  *float64        `json:"presence_penalty_penalty"`
 	TopP             *float64        `json:"top_p"`
 	ResponseFormat   *ResponseFormat `json:"response_format"`
+	Tools            []api.Tool      `json:"tools"`
```

Comment:
I believe you can just make this `any` and pass it through

Reply:
Not in this case, any would be incompatible with line 396 `Tools:    r.Tools,`